### ISSUE TITLE
Bug: Incorrect handling of quoted fields

### ISSUE THREAD

**Issue by ad-si** [BrianHicks/elm-csv#24]
```elm
> import Csv.Parser exposing (parse)

> parse { fieldSeparator = ',' } "Name,Species\nAtlas,cat\nAxel,dog\n"
Ok [["Name","Species"],["Atlas","cat"],["Axel","dog"]]
    : Result Csv.Parser.Problem (List (List String))

> parse { fieldSeparator = ',' } "\"Name\",\"Species\"\n\"Atlas\",\"cat\"\n\"Axel\",\"dog\"\n"
Ok [["Name","Species"],["Atlas","cat"],["Axel","dog",""]]
    : Result Csv.Parser.Problem (List (List String))
```

When the fields are quoted, it wrongly adds an additional empty field at the end.

**Comment by BrianHicks** [comment]
how strange! Feel like making a PR? It's unlikely that I'll be able to get this fixed very quickly. :\

**Comment by BrianHicks** [comment]
Fixed and released as 3.0.3. Hope you have a nice day!
